pinecone/index_connection.go

```diff
@@ -996,8 +1003,307 @@ func (idx *IndexConnection) DescribeIndexStatsFiltered(ctx context.Context, meta
 	}, nil
 }
 
-func (idx *IndexConnection) query(ctx context.Context, req *data.QueryRequest) (*QueryVectorsResponse, error) {
-	res, err := (*idx.dataClient).Query(idx.akCtx(ctx), req)
+// StartImportResponse holds the response parameters for the StartImport method.
+//
+// Fields:
+//   - Id: The ID of the import process that was started.
+type StartImportResponse struct {
+	Id string `json:"id,omitempty"`
+}
+
+// StartImport imports data from a storage provider into an index. The uri parameter must start with the
+// schema of a supported storage provider. For buckets that are not publicly readable, you will also need to
+// separately configure a storage integration and pass the integration id.
+//
+// Returns a pointer to a StartImportResponse object with the import ID or an error if the request fails.
+//
+// Parameters:
+//   - ctx: A context.Context object controls the request's lifetime,
+//     allowing for the request to be canceled or to timeout according to the context's deadline.
+//   - uri: The URI of the data to import. The URI must start with the scheme of a supported storage provider.
+//   - integrationId: If your bucket requires authentication to access, you need to pass the id of your storage integration using this property.
+//     Pass nil if not required.
```

Comment:
I would need to add a new struct type that allows batching `uri`, `integrationId`, and `errorMode`. I could change it but I felt like since there are so few fields at the moment maybe it's more overhead to have to worry about creating the struct rather than just passing values. Do you feel strongly about this?

Reply:
no def not, you do you!